### PR TITLE
Adapted bootstrap to be backwards compatible between old and new names

### PR DESCRIPTION
**Pull request by andrei-tyk**
## Description

We have renamed our "Tyk Enterprise Portal" chart to "Tyk Developer Portal". This PR aims to add the required changes to the environment variables to allow for the new changes and for backward compatibility.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] Make sure you are updating [CHANGELOG.md](../CHANGELOG.md) based on your changes.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `gofmt -s -w .`
  - [ ] `go vet ./...`
  - [ ] `golangci-lint run`

### CODE REVIEW

File: constants/constants.go [hunk 1]
```diff
@@ -3,10 +3,12 @@ package constants
 const (
 	OperatorSecretEnabledEnvVar         = "OPERATOR_SECRET_ENABLED"
 	EnterprisePortalSecretEnabledEnvVar = "ENTERPRISE_PORTAL_SECRET_ENABLED"
```
Comment (buraksekili) [inline constants/constants.go]: I do not think we should continue maintaining anything related to `EnterprisePortalSecretEnabledEnvVar` old naming of the Developer Portal chart.

File: helpers/operator.go [hunk 2]
```diff
@@ -88,32 +88,41 @@ func BootstrapTykEnterprisePortalSecret() error {
...
-			fmt.Println("A previously created enterprise portal secret was identified and deleted")
+			fmt.Println("A previously created portal secret was identified and deleted")
 			break
 		}
 	}
 
-	err = CreateTykEnterprisePortalSecret(clientset)
-	if err != nil {
-		return err
+	if data.AppConfig.EnterprisePortalSecretName != "" {
```
Comment (buraksekili) [inline helpers/operator.go]: If both `EnterprisePortalSecretName` and `DeveloperPortalSecretName` are populated with some values, we'll create both secrets. I think we should only go with the new name: `DeveloperPortalSecretName`

**Review by buraksekili — CHANGES_REQUESTED** [review]
Thank you, @andrei-tyk. I've added my comments and would appreciate your thoughts on them. Additionally, could you kindly update the PR description and details?

File: helpers/operator.go (hunk 2 above)
Comment (andrei-tyk) [reply helpers/operator.go]: Given that this is controlled at chart level, shouldn't we leave the option to use both?

File: constants/constants.go (hunk 1 above)
Comment (andrei-tyk) [reply constants/constants.go]: If we change this now then we would need to make changes to the old charts too.

Comment (buraksekili) [reply constants/constants.go]: I mean old charts will use releases with `v1.x.y`, for instance, v1.5.0 where the script already uses old names for Tyk Dev Portal. This change only required in the new charts, which will be released through v2.x.y as you suggested

File: helpers/operator.go (hunk 2 above)
Comment (buraksekili) [reply helpers/operator.go]: Once we merge this code piece to the main branch, only new helm-charts, which only provides the new `DeveloperPortalSecretName` option, will use this code piece. so, in my opinion, no need to maintain two versions of the environment variables.

File: constants/constants.go (hunk 1 above)
Comment (andrei-tyk) [reply constants/constants.go]: Ok, I see your point now. Adapted.

File: helpers/operator.go (hunk 2 above)
Comment (andrei-tyk) [reply helpers/operator.go]: Makes sense, implemented.

**Review by buraksekili — APPROVED** [review]

**Pull request merged**